test/libs/oboeAsserter.js

```diff
@@ -16,398 +8,405 @@ if(isNode) {
+    this.whenGivenInputOneCharAtATime = function (input) {
+      var jsonSerialisedInput
 
-      if (typeof input == 'string') {
-        jsonSerialisedInput = input;
+      if (typeof input === 'string') {
+        jsonSerialisedInput = input
       } else {
-        jsonSerialisedInput = JSON.stringify(input);
+        jsonSerialisedInput = JSON.stringify(input)
       }
 
-      if( !jsonSerialisedInput || jsonSerialisedInput.length == 0 ) {
-        throw new Error('Faulty test - input not valid:' + input);
+      if (!jsonSerialisedInput || jsonSerialisedInput.length === 0) {
+        throw new Error('Faulty test - input not valid:' + input)
       }
 
-      for( var i = 0; i< jsonSerialisedInput.length; i++) {
-        oboeInstance.emit('data', jsonSerialisedInput.charAt(i) );
+      for (var i = 0; i < jsonSerialisedInput.length; i++) {
+        oboeInstance.emit('data', jsonSerialisedInput.charAt(i))
       }
 
-      return this;
-    };
-
-    this.whenInputFinishes = function() {
+      return this
+    }
 
-      oboeInstance.emit('end');
+    this.whenInputFinishes = function () {
+      oboeInstance.emit('end')
 
-      return this;
-    };
+      return this
+    }
 
     /**
      * Assert any number of conditions were met on the spied callback
      */
-    this.thenTheInstance = function( /* ... functions ... */ ){
-
-      if( givenErrors.length > 0 ) {
-        throw new Error('error found during previous stages\n' + givenErrors[0].stack);
+    this.thenTheInstance = function (/* ... functions ... */) {
+      if (givenErrors.length > 0) {
+        throw new Error('error found during previous stages\n' + givenErrors[0].stack)
       }
 
       for (var i = 0; i < arguments.length; i++) {
-        var assertion = arguments[i];
-        assertion.testAgainst(spiedCallback, oboeInstance, completeJson);
+        var assertion = arguments[i]
+        assertion.testAgainst(spiedCallback, oboeInstance, completeJson)
       }
 
-      return this;
-    };
+      return this
+    }
 
     /** sinon stub is only really used to record arguments given.
      *  However, we want to preserve the arguments given at the time of calling, because they might subsequently
      *  be changed inside the parser so everything gets cloned before going to the stub
      */
-    function argumentClone(delegateCallback) {
-      return function(){
-
-        function clone(original){
-          // Note: window.eval being used here instead of JSON.parse because
-          // eval can handle 'undefined' in the string but JSON.parse cannot.
-          // This isn't wholy ideal since this means we're relying on JSON.
-          // stringify to create invalid JSON. But at least there are no
-          // security concerns with this being a test.
-          return globalContext.eval( '(' + JSON.stringify( original ) + ')' );
+    function argumentClone (delegateCallback) {
+      return function () {
+        function clone (obj) {
```

Comment:
Change here to the clone function. The current one drops undefined properties and removes types so e.g. an Error would become an Object. The type needed to be preserved for the instanceOf Error check to work.

Reply:
Agreed. I was operating on the basis that we couldn't use dependencies. With webpack we should be able to use just the bits we need so there should be too much in the way of bloat.

Reply:
I think it's probably fine to use whatever on the tests, though I've gotten the same impression as you generally 😄